source/security/authentication.txt

```diff
@@ -0,0 +1,401 @@
+method to authenticate, create a user with an associated Amazon Resource
+Name (ARN) on the ``$external`` database. Then, specify ``'MONGODB-AWS'`` as the value of
+the ``authMechanism`` connection option.
+
+When you use the MONGODB-AWS mechanism, the {+php-library+} tries to retrieve AWS
+credentials from the following sources, in the order listed:
+
+1. Options parameter passed to the ``MongoDB\Client`` constructor or parameters in the
+   connection URI
+#. Environment variables
+#. AWS EKS ``AssumeRoleWithWebIdentity`` request
+#. ECS container metadata
+#. EC2 instance metadata
+
+The following sections describe how to use the {+php-library+} to retrieve credentials from
+these sources and use them to authenticate your application.
+
+.. _php-mongodb-aws-credentials:
+
+MongoDB\Client Credentials
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, the {+php-library+} checks whether you passed AWS credentials
+to the ``MongoDB\Client`` constructor, either as an options parameter or as part of the
+connection URI. To pass your credentials to ``MongoDB\Client``,
+set the following connection options:
+
+- ``username``: The AWS IAM access key ID to authenticate. Percent-encode this value
+  before including it in a connection URI.
+- ``password``: The AWS IAM secret access key. Percent-encode this value before including
+  it in a connection URI.
+- ``authMechanism``: Set to ``'MONGODB-AWS'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $options = [
+             'username' => '<AWS IAM access key ID>',
+             'password' => '<AWS IAM secret access key>',
+             'authMechanism' => 'MONGODB-AWS',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<AWS IAM access key ID>:<AWS IAM secret access key>@<hostname>:<port>/?authMechanism=MONGODB-AWS';
+         $client = new Client($uri);
+
+.. _php-mongodb-aws-env-vars:
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+If you don't provide a username and password when you construct your ``MongoDB\Client``
+object, the {+php-library+} tries to retrieve AWS credentials from the following
+environment variables:
+
+- ``AWS_ACCESS_KEY_ID``
+- ``AWS_SECRET_ACCESS_KEY``
+- ``AWS_SESSION_TOKEN``
+
+To use these environment variables to authenticate your application, first set them to the
+AWS IAM values needed for authentication, as shown in the following code
+example:
+
+.. code-block:: sh
+
+   export AWS_ACCESS_KEY_ID=<AWS IAM access key ID>
+   export AWS_SECRET_ACCESS_KEY=<AWS IAM secret access key>
+   export AWS_SESSION_TOKEN=<AWS session token>
```

Comment:
A shell script would certainly work if PHP is being launched as a CLI process, but I think this is less relevant for web applications. Most frameworks have their own method for managing env vars (e.g. [phpdotenv](https://github.com/vlucas/phpdotenv)).

[Symfony](https://symfony.com/doc/current/configuration.html#env-file-syntax) has its own implementation, and Laravel uses [phpdotenv](https://github.com/vlucas/phpdotenv) directly. The `.env` file should look the same for both frameworks:

```
AWS_ACCESS_KEY_ID=<AWS IAM access key ID>
AWS_SECRET_ACCESS_KEY=<AWS IAM secret access key>
AWS_SESSION_TOKEN=<AWS session token>
```

You can also wrap the values in double quotes if you like, although I believe that's only necessary if they contain spaces (not sure how relevant that is for these secrets).

It may make sense to present two tabs here. The shell example using `export` and a `.env` file.